docs(components): update documentation on components (#1583)
AmadeusITGroup/otter: docs/components/CONTAINER_PRESENTER.md

Reviewer: ```suggestion
We encourage developers to decouple components into containers and presenters. From a UI perspective it is a good practice to separate access of data/business logic form pure presentation. This allows developers to reuse presenters in other parts of the code with different data or having a container linked to multiple presenters, in case you want to display the same thing with a totally different user experience.
```

> 

Reviewer 2: line 16: italics need to be replaced with `` ``

| **Component file name**      | *-cont.component.ts |
| **Selector name**      | *-cont |
| **Component name**      | *ContComponent |
| **Configuration file name**      | *-cont.config.ts |
| **Configuration name**      | *ContConfig |
| **Context file name**      | *-cont.context.ts |
| **Context interface names**      | *ContContextInput / *ContContextOutput / *ContContext |
| **Fixture file name**      | *-cont.fixture.ts |
| **Fixture name**      | 	*ContFixture |
| **Module file name**      | *-cont.module.ts |
| **Module name**      | *ContModule |
| **Template file name**      | *-cont.template.html |
| **Unit test file name**      | *-cont.spec.ts |

> It has its own _index.ts_ file exporting:

Author: It is pertinent in the documentation ?
What does it bring to the developer that we write it here ?

> It has its own _index.ts_ file exporting:

Reviewer 3: The fact that we export the different file of the components in a barrel, that does not include all of then per default, soon to be an information that is meaningful.

> The presenter is located in the _presenter_ folder of the component.

Reviewer 2: ```suggestion
The presenter is located in the __presenter__ folder of the component.
```

> ## Guidelines summary

Reviewer 2: Should we move it at the beginning of the page ? People will first care if it fits their use case before the implementation details

Reviewer 2: Erratum: seems more just guidelines or advices than a summary -- we can probably leave it here, but since we don't recommend it for all the use cases, we might need to make it clearer in the introduction

## Guidelines summary

* A component interacting with external entities (store, services, etc.) may be split into a container and a presenter

> * Subcomponents are located in the related block subcomponents folder. If the subcomponent is a block, it will then be set either in the components root folder (if it has a functional meaning), or at the root of the functional folder in a dedicated sub folder.

Reviewer: ```suggestion
* Subcomponents are located in the subcomponents folder of the related block. If the subcomponent is a block, it will then be set either in the component's root folder (if it has a functional meaning), or at the root of the functional folder in a dedicated subfolder.
```

> * A component shared among different blocks and hosted in the same functional folder, should be placed in a shared folder under the functional one.

Reviewer: ```suggestion
* A component shared among different blocks and hosted in the same functional folder should be placed in a shared folder under the functional one.
```

```
app/src/
  components/
    my-complex-area/
      my-complex-component/
        container/
        contracts/
        presenter/
        sub-components/
          my-sub-component/
      my-simple-component/
      shared/
        my-shared-component-in-my-complex-area/
          container/
          presenter/
    my-simple-area/
      container/
      presenter/
    shared/
      my-shared-component-in-app/
        container/
        presenter/
  elements/
    my-element/

> ```

Author: added while resolving https://github.com/AmadeusITGroup/otter/pull/1583/files#r1558755698
